pkg/cmd/cli/config/smart_merge.go

```diff
@@ -67,42 +124,28 @@ func MergeConfig(haystack, addition clientcmdapi.Config) (*clientcmdapi.Config,
 	ret := haystack
 
 	requestedClusterNamesToActualClusterNames := map[string]string{}
-	existingClusterNames, err := getMapKeys(reflect.ValueOf(haystack.Clusters))
-	if err != nil {
-		return nil, err
-	}
 	for requestedKey, needle := range addition.Clusters {
 		if existingName := FindExistingClusterName(haystack, needle); len(existingName) > 0 {
```

Comment:
do we still want to scan/reuse cluster stanzas, or only use the stanza name `getClusterNickname` would produce?